gcc/config/riscv/riscv.md

```diff
@@ -1718,7 +1718,7 @@
 (define_insn "zero_extendqi<SUPERQI:mode>2"
   [(set (match_operand:SUPERQI 0 "register_operand"    "=r,r")
 	(zero_extend:SUPERQI
-	    (match_operand:QI 1 "nonimmediate_nonpostinc" " r,m")))]
+	    (match_operand:QI 1 "nonimmediate_nonpostinc" " r,am")))]
```

Comment:
Selectively changing some of the `m`'s to `am` seems haphazard. I think base reg + index reg and post modify should be the exceptions and `m` the norm. Have a look at `TARGET_MEM_CONSTRAINT`.